ssh/ssh.go

```diff
@@ -20,22 +21,71 @@ type Key struct {
 	Fingerprint string // Name to associate with fingerprint
 }
 
-// GenerateKey will create a private key and fingerprint
-func GenerateKey(secrets *v1.Secret, key Key) {
+// RecordKeyInfo records priave key or fingerprint names for later generation
+func RecordKeyInfo(keys map[string]Key, configVar *model.ConfigurationVariable) error {
+	if len(configVar.Generator.ID) == 0 {
+		return fmt.Errorf("Config variable `%s` has no ID value", configVar.Name)
+	}
+	if configVar.Generator.Type != model.GeneratorTypeSSH {
+		return fmt.Errorf("Config variable `%s` does not have a valid SSH generator type", configVar.Name)
+	}
+
+	// Get or create the key from the map, there should always be
+	// a pair of private keys and fingerprints
+	key := keys[configVar.Generator.ID]
+
+	switch configVar.Generator.ValueType {
+	case model.ValueTypeFingerprint:
+		if len(key.Fingerprint) > 0 {
```

Comment:
Using `== ""` here might be clearer (but both work fine)

Reply:
idk, we once changed all these comparisons with the empty string to a `len` check (I think based on Aaron's feedback, but not sure). We could change back to compare against strings, but then we should do it everywhere (and there are a ton of these checks in this codebase).

Reply:
Yeah, no need to change it now. I think I remember that (and checking the assembly output to see if gc was smart enough to optimize empty string checks… I think it wasn't?)